plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -366,13 +374,37 @@ public void setTablePrincipal(ConnectorSession session, SchemaTableName schemaTa
     @Override
     public void createView(ConnectorSession session, SchemaTableName schemaViewName, ConnectorViewDefinition definition, boolean replace)
     {
-        throw new TrinoException(NOT_SUPPORTED, "createView is not supported for Iceberg REST catalog");
+        ViewBuilder builder =
+                restSessionCatalog.buildView(convert(session), toIdentifier(schemaViewName));
+        Schema schema = IcebergUtil.schemaFromViewColumns(typeManager, definition.getColumns());
+        Namespace defaultNamespace = Namespace.of(definition.getSchema().orElse(schemaViewName.getSchemaName()));
+        Map<String, String> properties = new HashMap<>();
+        if (definition.isRunAsInvoker()) {
+            properties.put(TRINO_RUN_AS_INVOKER, "true");
+        }
+
+        String tableLocation = defaultTableLocation(session, schemaViewName);
+        definition.getOwner().ifPresent(owner -> properties.put(TRINO_OWNER, owner));
+
+        ViewBuilder viewBuilder = builder.withSchema(schema)
+                .withQuery(TRINO_SQL_DIALECT, definition.getOriginalSql())
+                .withDefaultNamespace(defaultNamespace)
+                .withDefaultCatalog(definition.getCatalog().orElse(catalogName.toString()))
```

Comment:
as above -- the ConnectorViewDefinition retrieved later on must have `ConnectorViewDefinition.catalog` empty if the `ConnectorViewDefinition` being stored had `ConnectorViewDefinition.catalog` empty .